Orange/widgets/data/owgroupby.py

```diff
@@ -0,0 +1,490 @@
+        time. In this case we step between unchecked, partially checked and
+        checked or just between unchecked and checked - depending on situation
+        """
+        if self.checkState() == Qt.Checked:
+            # if checked always uncheck
+            self.setCheckState(Qt.Unchecked)
+        else:
+            agg = self.text()
+            selected_attrs = self.parent.get_selected_attributes()
+            types_ = set(type(attr) for attr in selected_attrs)
+            can_be_applied_all = (AGGREGATIONS[agg].types & types_) == types_
+
+            # true if aggregation applied to all attributes that can be
+            # aggregated with selected aggregation
+            applied_all = all(
+                type(attr) not in AGGREGATIONS[agg].types
+                or agg in self.parent.aggregations[attr]
+                for attr in selected_attrs
+            )
+            if self.checkState() == Qt.PartiallyChecked:
+                # if partially check: 1) check if agg can be applied to all
+                # 2) else uncheck if agg already applied to all
+                # 3) else leve partially checked to apply to all that can be aggregated
+                if can_be_applied_all:
+                    self.setCheckState(Qt.Checked)
+                elif applied_all:
+                    self.setCheckState(Qt.Unchecked)
+                else:
+                    self.setCheckState(Qt.PartiallyChecked)
+                    # since checkbox state stay same signal is not emitted
+                    # automatically but we need a callback call so we emit it
+                    self.stateChanged.emit(Qt.PartiallyChecked)
+            else:  # self.checkState() == Qt.Unchecked
+                # if unchecked: check if all can be checked else partially check
+                self.setCheckState(
+                    Qt.Checked if can_be_applied_all else Qt.PartiallyChecked
+                )
+
+
+@contextmanager
+def block_signals(widget):
+    widget.blockSignals(True)
+    yield
+    widget.blockSignals(False)
+
+
+class OWGroupBy(OWWidget, ConcurrentWidgetMixin):
+    name = "Group by"
+    description = ""
+    icon = "icons/GroupBy.svg"
+    keywords = ["aggregate", "group by"]
+
+    class Inputs:
+        data = Input("Data", Table, doc="Input data table")
+
+    class Outputs:
+        data = Output("Data", Table, doc="Aggregated data")
+
+    class Error(OWWidget.Error):
+        unexpected_error = Msg("{}")
+
+    settingsHandler = DomainContextHandler()
+
+    gb_attrs: List[Variable] = ContextSetting([])
+    aggregations: Dict[Variable, Set[str]] = ContextSetting({})
+    auto_commit: bool = Setting(True)
+
+    def __init__(self):
+        super().__init__()
+        ConcurrentWidgetMixin.__init__(self)
+
+        self.data = None
+        self.result = None
+
+        self.gb_attrs_model = DomainModel(
+            valid_types=(DiscreteVariable, ContinuousVariable, StringVariable),
```

Comment:
Why prohibit `TimeVariable`? It is just as discrete as `ContinuousVariable`.

Reply:
It was already included since it is a subclass of ContinuousVariable. For the purpose of code readability, I added it.

Reply:
If all types are valid, I'd suggest to increase the readability by omitting the parameter. :) Allowing all is what `DomainModel` does by default.

Reply:
`valid_types` is redundant since we include all variables -- I removed it. It was here since I probably want to make aggregation differently in the beginning.